### PR TITLE
[camera] Fix iOS rotation issue

### PR DESCRIPTION
**Pull request by mvanbeusekom**
This PR fixes an issue where on iOS the default orientation was set to landscape right while it should have been set to portrait up. To compensate for this problem the `CameraPreview` widget used an additional offset when calculating the rotation on iOS. 

This PR solves the problem at the root and also removed the extra offset needed on iOS to adjust the orientation.

- Fixes flutter/flutter#76210

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Review by bparrishMines — COMMENTED** [review]
Everything looks good, but this still lacks tests. Potentially a widget test?

**Comment by mvanbeusekom** [comment]
@bparrishMines on iOS this indeed looks good but on Android I am running into the following problem: https://discord.com/channels/608014603317936148/608020293944082452/815873332574158890 (if you have a solution or advise I would love to hear it)

I will mark the PR as draft until this gets resolved. And I will also make sure to add tests.

File: packages/camera/camera/lib/src/camera_preview.dart [hunk 1]
```diff
@@ -41,26 +40,25 @@ class CameraPreview extends StatelessWidget {
...
       DeviceOrientation.portraitUp: 0,
       DeviceOrientation.landscapeLeft: 1,
       DeviceOrientation.portraitDown: 2,
       DeviceOrientation.landscapeRight: 3,
     };
-    return turns[_getApplicableOrientation()]! + platformOffset;
+    return turns[_getApplicableOrientation()]!;
+  }
+
+  DeviceOrientation _getApplicableOrientation() {
```
Comment (cyanglaz) [inline packages/camera/camera/lib/src/camera_preview.dart:58]: Just to confirm, there is no diff in this method right? The method was just moved down?

Comment (mvanbeusekom) [reply packages/camera/camera/lib/src/camera_preview.dart:58]: Correct, it made more sense to me to keep it closer to where it was used. I see now that it causes a but of noice

Comment (cyanglaz) [reply packages/camera/camera/lib/src/camera_preview.dart:58]: keeping it close to where it was used makes sense to me!

**Comment by mvanbeusekom** [comment]
> Everything looks good, but this still lacks tests. Potentially a widget test?

@bparrishMines I have added some widget tests to make sure the rotated box turns correctly according to the different settings provided by the `CameraController`. Would appreciate it is you could have another look.

**Comment by bparrishMines** [comment]
@mvanbeusekom As stated on Discord, Camera2 on Android automatically changes the orientation of the output while iOS requires you to set the value. 

On iOS, have you tried listening to the the [UIDeviceOrientationDidChangeNotification](https://developer.apple.com/documentation/uikit/uideviceorientationdidchangenotification?language=objc) and automatically updating the output camera preview orientation? I don't think there's anything that can be done on Android about Orientation since we use a `SurfaceTextureEntry` and not a `TextureView`.

**Comment by mvanbeusekom** [comment]
> On iOS, have you tried listening to the the [UIDeviceOrientationDidChangeNotification](https://developer.apple.com/documentation/uikit/uideviceorientationdidchangenotification?language=objc) and automatically updating the output camera preview orientation?

Yes that is exactly what I did in this pull request, which we tested extensively and solves all orientation issues on iOS without the need of a `RotatedBox` on the Dart side. This is now ready and added additional automation tests to ensure this is working properly. 

I was just wondering if we could do something similar on Android so we could remove the `RotatedBox` on the Dart side altogether. I had an offline discussion with @cyanglaz and we agreed to land this PR first fixing the issues on iOS (which are currently the biggest problem) and file a new feature request for the Flutter Engine to see if its possible to expose the `TextureView` in the future so we could also solve this issue in a more elegant way for Android (without the need for additional coding on the Dart side).

So in short, this PR should be ready and would be great if you could review it.

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by snoopdoggy322** [comment]
In my case, updating to a new version did not fix this problem
Device: iPhone SE(old)

https://user-images.githubusercontent.com/38835822/122252946-67839f80-ced4-11eb-89c3-787586fbfcd7.MP4
